### DISCUSSION TITLE
Setting up a simple Dear ImGui project from scratch

### DISCUSSION THREAD

**Discussion by TomyFrank** [ocornut/imgui#6931] [New Users Build/Link/Run issues ONLY!]
I've compiled the lastest version of Dear ImGui (docking branch) on my VS 2022 on Windows suceessfully. The examples work fine too. As begginer of using Dear ImGui I'm interested in creating a project from scratch. I tried much to take advantages of Getting started, FAQ and Wiki sections in addition to Google as much as I could, for that purpose, yet am not able to approprately make a project from scratch using this libray which makes it possible to create GUI projects using C++ fast and easily.

The project is very simple and the purpose is mostly learning how to use Dear ImGui better. Suppose we need a very simple project that merely shows a circle and a line crossing it. First, on VS configuration settings, I added the path to _ImGui\MyApplication\backends\examples\libs\glfw\include_ and _ImGui\MyApplication\backends_ to the project _properties -> Additional Include Directories_ and in _Linker -> General -> Additional Library Directories_ added  _\ImGui\MyApplication\examples\libs\glfw\lib-vc2010-64_.
The code in the image below addresses must of the purpose to me but yet I get a couple of errors.

![undefined - Imgur](https://github.com/ocornut/imgui/assets/147997682/72c4b9c4-64ed-4c8f-80b5-e2308b724737)

How to fix those errors, or, how would you create such a project from scratch, please?

**Comment by ocornut** [comment]
This is mostly a C++ usage question. We cannot normally provide this sort of support here as this is way outside our scope.
You should be comparing your code to the provided examples to understand and study the differences.

Some hint through:
- you need to include glfw3.h to get the declaration of GLFW functions such as glfwWindowShouldClose().
- you are not creating a GLFW window: you need to. compare your code to example.
- you need to add your imgui/ folder to include paths in order to use `#include "imgui.h"`. presumably imgui.h itself includes imconfig.h and expects all files to be in the same location.
